Some options are not present in illumos grep, use GNU grep instead.

--- zstd-1.4.0/programs/zstdgrep	2019-04-17 00:37:24.000000000 +0000
+++ zstd-1.4.0/programs/zstdgrep.new	2019-05-11 14:09:27.963420412 +0000
@@ -22,7 +22,7 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-grep=${GREP:-grep}
+grep=${GREP:-ggrep}
 zcat=${ZCAT:-zstdcat}
 
 endofopts=0
